### PR TITLE
Validator processor

### PR DESCRIPTION
**Pull request by krzlabrdx**

### CODE REVIEW

File: src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/ValidatorProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    }
+
+    public void VisitUpsert(CoreModel.Substate substateData, ReferencedEntity referencedEntity, long stateVersion, long? passingEpoch)
+    {
+        if (substateData is CoreModel.ConsensusManagerRegisteredValidatorsByStakeIndexEntrySubstate entry)
+        {
+            var av = entry.Value.ActiveValidator;
+            var lookup = new ValidatorPublicKeyLookup(_referencedEntities.Get((EntityAddress)av.Address).DatabaseId, av.Key.KeyType.ToModel(), av.Key.GetKeyBytes());
+
+            _seenPublicKeys.TryAdd(lookup, stateVersion);
```
Comment (PawelPawelec-RDX) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/ValidatorProcessor.cs:110]: why do we scan two different states for public keys? are both of them allowed to mention on ledger that public key for first time ?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/ValidatorProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        }
+
+        if (substateData is CoreModel.ConsensusManagerFieldCurrentValidatorSetSubstate validatorSet)
+        {
+            if (!passingEpoch.HasValue)
+            {
+                throw new InvalidOperationException("ConsensusManagerFieldCurrentValidatorSetSubstate can't be processed unless epoch change gets detected");
+            }
+
+            var activeSet = new Dictionary<ValidatorPublicKeyLookup, TokenAmount>();
```
Comment (PawelPawelec-RDX) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/ValidatorProcessor.cs:127]: that name is missleading, can we rename it to i.e validatorStakeDictionary? or sthn else that would indicate we hold stake amount per validator here?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/ValidatorProcessor.cs [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+internal record ValidatorActiveSet(long Epoch, IDictionary<ValidatorPublicKeyLookup, TokenAmount> ValidatorSet, long StateVersion);
+
+internal class ValidatorProcessor
+{
+    private readonly ProcessorContext _context;
+    private readonly ReferencedEntityDictionary _referencedEntities;
+
+    private List<ValidatorActiveSet> _changes = new();
+
+    private Dictionary<ValidatorPublicKeyLookup, long> _seenPublicKeys = new();
```
Comment (PawelPawelec-RDX) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/ValidatorProcessor.cs:91]: we have 3 collections for public keys here, it's not clear for me what is responsibility of each of them. Can we rename them or even better introduce some abstraction layer here to make it clear?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/ValidatorProcessor.cs (hunk 1 above)
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/ValidatorProcessor.cs]: Clarified.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/ValidatorProcessor.cs (hunk 2 above)
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/ValidatorProcessor.cs]: Clarified.

**Review by PawelPawelec-RDX — APPROVED** [review]

**Pull request merged**